remote/rpc/fallback.go

```diff
@@ -0,0 +1,86 @@
+package rpc
+
+import (
+	"context"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/weaveworks/flux/api"
+	"github.com/weaveworks/flux/api/v6"
+	"github.com/weaveworks/flux/api/v9"
+	"github.com/weaveworks/flux/job"
+	"github.com/weaveworks/flux/remote"
+	"github.com/weaveworks/flux/update"
+)
+
+// fallback is, as the name suggests, a fallback api.UpstreamServer implementation.
+// Any method not implemented in a given RPC client will terminate here.
+type fallback struct{}
+
+type completeServer interface {
```

Comment:
This will make failure to implement a "meaningful" `Close` method into a runtime error. Wouldn't it be better if it were a compile time error? I'm not sure why we have `fallback` (formerly `baseError`), any more.

Reply:
> This will make failure to implement a "meaningful" Close method into a runtime error.

Yes it will, good point. I shall rethink this.

> I'm not sure why we have fallback (formerly baseError), any more.

I'll look into the flux-api code implications for removing the fallback at some point. My suspicion is some code duplication will arise, but hopefully not.